prow/tide/tide.go

```diff
@@ -575,9 +629,9 @@ func (c *Controller) initSubpoolData(sp *subpool) error {
 	}
 	sp.cc = make(map[int]contextChecker, len(sp.prs))
 	for _, pr := range sp.prs {
-		sp.cc[int(pr.Number)], err = c.config().GetTideContextPolicy(c.gc, sp.org, sp.repo, sp.branch, refGetterFactory(string(sp.sha)), string(pr.HeadRefOID))
+		sp.cc[pr.Number], err = c.config().GetTideContextPolicy(c.gc, sp.org, sp.repo, sp.branch, refGetterFactory(string(sp.sha)), pr.HeadRefOID)
 		if err != nil {
-			return fmt.Errorf("error setting up context checker for pr %d: %w", int(pr.Number), err)
+			return fmt.Errorf("error setting up context checker for pr %d: %w", pr.Number, err)
```

Comment:
FWIW we'll probably need to include the initSubpoolData function in the controller interface since this will be different for Gerrit and GitHub. The use of the `sp.presubmits` map could be consistent between the two, but the generation of the map (via `presubmitsByPull`) will likely need to consider Gerrit reporting labels and the context policy checker only applies to GitHub.